File: examples/c/document_new_from_string.c
#include <gxml/gxml.h>
#include <stdio.h>

int main () {
  char *xml;
  char *str;
  GXmlDocument *doc;

  xml = "<?xml version=\"1.0\"?><Bookshelf><Owner fullname=\"John Green\"/><Books><Book author=\"John Green\" title=\"The Fault in Our Stars\"/><Book author=\"Jane Austen\" title=\"Pride &amp; Prejudice\"/><Book author=\"J.D. Salinger\" title=\"Nine Stories\"/></Books></Bookshelf>";

  doc = gxml_document_new_from_string (xml);

  str = gxml_node_to_string (GXML_NODE (doc), TRUE, 2);
  printf ("%s:\n%s\n", __FILE__, str);
  g_free (str);

  g_object_unref (doc);

  return 0;
}

File: examples/js/document_new_from_string.js
#!/usr/bin/gjs

const GXml = imports.gi.GXml;
const Gio = imports.gi.Gio;

function create_a_document_from_a_string () {
    let xml = "<?xml version=\"1.0\"?>\
<Bookshelf>\
<Owner fullname=\"John Green\"/>\
<Books>\
<Book author=\"John Green\" title=\"The Fault in Our Stars\"/>\
<Book author=\"Jane Austen\" title=\"Pride &amp; Prejudice\"/>\
<Book author=\"J.D. Salinger\" title=\"Nine Stories\"/>\
</Books>\
</Bookshelf>";
    let doc = GXml.GomDocument.from_string (xml);
    print ("create_a_document_from_a_string:\n" + doc.write_string ());
    
}

create_a_document_from_a_string ();
